File: make_vers.rb
require "date.rb"
fbuild = IO.readlines('build')
build_pars = fbuild.at(0).split('_')
xFile = File.new('VERSION', 'w')
build_num = build_pars.at(1).to_i + 1
stroka = 'Tab_calc version ' + build_pars.at(0) + ' build ' + build_num.to_s+' ('+Date.today.to_s+')'
xFile.puts stroka
xFile.puts 'Written by A.A.Mints'
xFile.close


xFile = File.new('build', 'w')
xFile.puts build_pars.at(0)+'_'+build_num.to_s
xFile.close

File: make_vers.py
#!/usr/bin/python
from datetime import datetime
fbuild = open('build', 'r').readlines()
build_pars = fbuild[0].split('_')
xFile = open('VERSION', 'w')
build_num = int(build_pars[1]) + 1
xFile.write('Tab_calc version %s build %s (%s)\n' % (build_pars[0], build_num, str(datetime.now())))
xFile.write('Written by A.A.Mints')
xFile.close()

xFile = open('build', 'w')
xFile.write(build_pars[0]+'_'+str(build_num))
xFile.close()
